app/assets/javascripts/models/ui/spatial_type.js.coffee

```diff
@@ -7,25 +7,76 @@ ns.SpatialType = do (ko, $=jQuery) ->
     constructor: (@query) ->
       @icon = ko.observable('fa-crop')
       @name = ko.observable('Spatial')
+      @displaySpatial = ko.computed(read: @_getDisplaySpatialName, write: @_clearDisplaySpatialName, owner: this, deferEvaluation: true)
+      @querySpatial = ko.computed(read: @_getQuerySpatial, owner: this, deferEvaluation: true)
+      @mapControlTop = ko.computed(read: @_setMapControlPosition, owner: this)
+      @manualEntryVisible = ko.observable(false)
+
+    clearManualEntry: =>
+      @manualEntryVisible(false)
 
     selectNone: =>
       @name('Spatial')
       @icon('fa-crop')
+      if @displaySpatial() then @manualEntryVisible(true) else @manualEntryVisible(false)
 
     selectPoint: =>
       @name('Point')
       @icon('fa-map-marker')
+      @manualEntryVisible(true)
 
     selectRectangle: =>
       @name('Rectangle')
       @icon('edsc-icon-rect-open')
+      @manualEntryVisible(true)
 
     selectPolygon: =>
       @name('Polygon')
       @icon('edsc-icon-poly-open')
+      @manualEntryVisible(false)
 
     selectShapefile: =>
       @name('Shape File')
       @icon('fa-file-o')
+      @manualEntryVisible(false)
+
+    _toReadableName: (name)->
+      return 'Point' if name == 'point'
+      return 'Rectangle' if name == 'bounding_box'
+      null
+
+    _getDisplaySpatialName: =>
+      spatialParam = @_toReadableName(@query.spatial()?.split(':')[0])
+      if @name() == 'Spatial'
+        # Order matters
+        if spatialParam && @name() != spatialParam && @displaySpatial() != spatialParam && @name() != @displaySpatial()
+          # on spatial type changes from one (e.g. point) to another (e.g. rectangle).
+          return @displaySpatial()
+#        else if spatialParam == null && @name() =='Spatial' && (@displaySpatial() == 'Point' || @displaySpatial() == 'Rectangle')
+#           on clearing filters
+#          return @name()
+        if spatialParam?.length > 0
+          @manualEntryVisible(true)
+          return spatialParam
+        return @displaySpatial() if @displaySpatial()
+#        return '' if spatialParam == null
+#      if spatialParam != @name() then return @name() else return @name()
+      if spatialParam then spatialParam else @name()
+
+    _clearDisplaySpatialName: (value) =>
```

Comment:
If I call a writer on a compute property, I expect the backend values to be set. This doesn't do anything. It looks like this is just a debug method?